.gitlab-ci.yml

```diff
@@ -105,8 +115,9 @@ include:
   - local: '/gitlab-pipeline/stage/init-mender-monitor-package.yml'
   - local: '/gitlab-pipeline/stage/build.yml'
   - local: '/gitlab-pipeline/stage/test.yml'
-  - local: '/gitlab-pipeline/stage/publish.yml'
-  - local: '/gitlab-pipeline/stage/trigger.yml'
+  - local: '/gitlab-pipeline/stage/codecov.yml'
```

Comment:
this I would keep at `publish` since it is a good practice to not tie names to a concrete service or implementation, especially considering the coveralls/ codecov change...
not sure if there are other artifacts we might publish, but if that is unlikely maybe `publish-tests` - otherwise I would stay consistent with the other repos (`publish` for packages, images & tests)?

Reply:
> this I would keep at `publish` since it is a good practice to not tie names to a concrete service or implementation, especially considering the coveralls/ codecov change...

That was an unfortunate coincidence, I wrote "codecov" as short for "code coverage" but I didn't realize that it is the name of one specific service...

> not sure if there are other artifacts we might publish, but if that is unlikely maybe `publish-tests` - otherwise I would stay consistent with the other repos (`publish` for packages, images & tests)?

While I think consistency is key across repositories, `mender-qa` one is where we can have exceptions. Here the "publishing" is happening at `release` stage. You could argue that therefore it makes sense to merge these two, keeping it under `publish` and to both code coverage publishing and release artifacts publishing - but I think it is better to be explicit, as the `release` part only happens during releases while the test coverage publish happens on regular PRs.


All that being said, I hear your concern! I will just move it to `publish-coverage` but keep the `release` independent.

Reply:
> All that being said, I hear your concern! I will just move it to `publish-tests` but keep the `release` independent.

Alternatively, we could just move the code publishing to the `after_script` of the acceptance tests jobs and remove this stage. Will that make sense?

In case you are not aware, we only publish coverage for client acceptance tests (or, generally speaking, yocto components acceptance tests) but not for integration or backend-tests.

What do you guys think about this alternative?

Reply:
I agree, this is better. The separate upload step has always struck me as a bit odd. You would expect that results from tests are uploaded from the same job as where the tests were run.

Reply:
Good. This is not the main goal of this PR, so I'll work on it later on (leaving for now the `codecov` stage as it is).